### PR TITLE
Establish legacy and experimental distribution keyboards in Keyboards repo

### PR DESCRIPTION
**Pull request by mcdurdin**
This is a very large PR. It's best to review it locally rather than within the GH UI. It is made up of several components:

1. A reorganisation of all existing keyboards into a release/ folder and further grouping by first letter or letter group
2. Addition of all other legacy desktop keyboards from the Tavultesoft keyboard distribution repository
3. Addition of all other legacy web keyboards from the s.keyman.com 'Keyman Cloud' distribution repository
4. Addition of a .keyboard_info file for every keyboard in the repository
5. Rewrite of build scripts to use a build.sh script in the top-level folder, and avoid having scripts in each folder.
6. Removal of all other build scripts (build.bat kept as a stub that calls build.sh).
7. Addition of latest Keyman Developer Compiler, which runs on Windows and WINE.
8. Removal of dependency on Keyman Developer (so this should be able to run on Win/Mac/Linux)

### CODE REVIEW

File: legacy/halqemeylem_u/halqemeylem_u.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (jahorton) [inline legacy/halqemeylem_u/halqemeylem_u.json]: Is this supposed to remain a .json file?

**Comment by mcdurdin** [comment]
>I'm also a bit curious - I noticed that the legacy version .json data was moved into the non-legacy keyboard folders as a .keyboard_info file? Would this conflict in any way with non-legacy keyboards if a legacy version existed for that keyboard?

The existing version .json data remains for now. The .keyboard_info data is the master data we will use for building the distribution database. At some point we will consider if the data can be merged, but the .json data is really targeted at ad-hoc distribution and so could probably come out of this repo in the future.

**Comment by mcdurdin** [comment]
* note: Still to go, merging changes from `master`. Out of time for this afternoon.

**Comment by LornaSIL** [comment]
**Regarding fonts:**
For the release\shared\fonts would you consider using [ISO 15924](http://www.unicode.org/iso15924/codelists.html) codes instead of whatever someone feels like calling the script? It would be better to be standardized in that area I think. Instead of what is there now it would be: Cans, Ethi, Sinh, Taml.

If the fonts will all be in one folder we need to rethink the files that go with them. For instance, the OFL.txt goes with AbyssinicaSIL-R.ttf but no one would know that without looking in it. I cannot tell which fonts are associated with the GNU-GPLv2.txt file.

It might be best for each font distribution to have it's own folder under the 15924 script code so that the licensing doesn't get mixed up. 

Also, do we need to do some thinking about whether we are legally allowed to redistribute these fonts? Certainly with some of the licenses we can. Some of the fonts don't have any license information and I would tend to think we either need to research the license or don't include them.

Our goal could be to have one very comprehensive font for each script in here and keyboards could just link to other fonts as suggestions. The problem with redistributing so many fonts is that if they are updated, we continue redistributing old versions.

**Comment by LornaSIL** [comment]
In the root folder:
* The README.md file will need enhancing to help people understand the folder structure
     * What does "legacy" mean and what goes in "release". In the Unicode world "legacy" means non-Unicode, but in this system I believe it means we don't necessarily have the source? This should be explained.
     * Explain about the dividing up alphabetically. Is it sorted by keyboard name or by keyboard folder?

* sil_ethiopic is an empty folder than can be deleted.
* el_harari_ethiopic and el_harari_ethiopic\source are empty folders that can be deleted.
* arabic_izza has other keyboards in it (which do not seem to be in the release folder) and the arabic_izza folder should be deleted.

**Comment by mcdurdin** [comment]
@Lorna thank you for the comments regarding fonts. Do you mind if we work on that as a separate PR? I've copied your comments over into issue #100 and will respond there.

I will update README.md.

**Comment by mcdurdin** [comment]
Note: changes to help.keyman.com/developer/keyboards will come through together with some other work that @jahorton is doing.

**Pull request merged**